app/controllers/usps_upload_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UspsUploadController < ApplicationController
+  def create
+    authorize do
+      UspsUploader.new.run unless HolidayService.observed_holiday?(today)
+      render plain: 'ok'
+    end
+  end
+
+  private
+
+  def authorize
+    # Check for empty to make sure that the token is configured
+    if authorization_token && authorization_token == Figaro.env.usps_upload_token
```

Comment:
This is reasonable. Until we have a larger set of api endpoints like this (essentially webhooks), it's hard to do anything that takes more effort.